sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -162,7 +160,7 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
       }
 
     case RefreshTable(r: ResolvedTable) =>
-      RefreshTableExec(r.catalog, r.identifier, invalidateCache(r, recacheTable = true)) :: Nil
+      RefreshTableExec(r.catalog, r.identifier, recacheTable(r)) :: Nil
```

Comment:
Is this correct? This looks wrong to me because we don't recache if the original table is not cached from the beginning. The new code looks like caching always.

cc @sunchao

Reply:
Please see `if (cache.isDefined) {` condition, @MaxGekk .

Reply:
> This looks wrong to me because we don't recache if the original table is not cached 

This does the same. If something wasn't in the cache, it will not appear there.

Reply:
Let's imagine, we have cached Table1, and cached View1 which uses Table1.
1. Current implementation: remove Table1 and View1, and add Table1 back. View1 will be uncached forever.
2. Proposed: clear cached data of Table1 and View1, and keep them cached, so, next action will fill the cache for both Table1 and View1.

Reply:
Yes I'm +1 on the change. We should change the PR description & JIRA title to properly reflect this though (currently it looks like a refactoring). We should also document this in SQL migration guide as well.

Also cc @cloud-fan for opinions on this.

Reply:
It was a refactoring till your test failed ;-)

Reply:
Ah, got it. I was confused at `recacheTable`. Thanks.